spec/src/main/asciidoc/config.asciidoc

```diff
@@ -0,0 +1,147 @@
+//
+// Copyright (c) 2019 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+[[config]]
+== Common Configuration Patterns
+
+Many MicroProfile specifications make use of the Configuration specification for
+setting or overriding behaviour characteristics. Listed below are a list of recommended usage
+patterns that are already adopted by one or more of the MicroProfile specifications.
+
+[[patterns]] 
+=== Patterns
+
+[[propertyPrefix]]
+==== Configuration Property Name Prefixes
+
+In order to prevent collision it is recommended that properties that are not
+named to match application class names are prefixed with "mp." followed by the specification's short name:  `mp.<spec-name>`.
+
+----
+// For example, the Messaging Specification has an @Outgoing annotation:
+   mp.messaging.outgoing.attributename=3
+----
+
+[[qualifyingClassNames]]
+==== Properties That Relate To Application Classes
+
+All references to users' application classes in property names should be fully package qualified.
+
+There are places where an annotation class name appears or a specification sourced class name (e.g. Context Propagation) appears in configuration property names, these class names are not package qualified. Should these start with `mp.<spec-name>`? In this case `mp.ft`?
```

Comment:
Did you mean for this to be resolved before delivering the changes, or to leave the question open for future discussion?

Reply:
My intention was that it would provoke discussion and be resolved before the PR was merged. If it is not resolved I will reword/remove this sort of language and this sentance in particular.

Reply:
I will add a commit that removes all these questions into seperate issues and resolve this conversation point when this is done.

Reply:
THe next commit will contain:
All references to users' application classes in property names should be fully package qualified.

There are places where an annotation class name appears or a specification sourced class name
appears. These, going forwards, should start with `mp.<spec-name>`? 
----
// For example: 
   mp.ft.retry.maxRetries=30
----

There are legacy configuration options that do not yet follow this convention
and these should be converged when convenient without breaking current configurations.
See for example: https://github.com/eclipse/microprofile-fault-tolerance/issues/435